x-pack/plugins/security_solution/server/usage/endpoints/index.ts

```diff
@@ -93,77 +96,93 @@ export const trackEndpointOSTelemetry = (
-        const userDesiredMalwareState =
-          endpointPolicyDetails['endpoint-security'].Endpoint?.configuration?.inputs[0]?.policy[
-            hostType
-          ]?.malware?.mode;
-
-        const isAnActiveMalwareState = enabledMalwarePolicyTypes.includes(userDesiredMalwareState);
-        const malwareStatus =
-          endpointPolicyDetails['endpoint-security'].Endpoint?.policy?.applied?.response
-            ?.configurations?.malware?.status;
-
-        if (isAnActiveMalwareState && malwareStatus !== 'failure') {
-          updatedEndpointTelemetry.policies.malware.active += 1;
-        }
-        if (!isAnActiveMalwareState) {
-          updatedEndpointTelemetry.policies.malware.inactive += 1;
-        }
-        if (isAnActiveMalwareState && malwareStatus === 'failure') {
-          updatedEndpointTelemetry.policies.malware.failure += 1;
-        }
-      }
-    }
+  if (!payload) {
+    // This payload may not always be provided depending on the state of the endpoint. Guard again situations where it is not sent
+    return policiesTracker;
+  }
+
+  let endpointPolicyPayload;
+  try {
+    endpointPolicyPayload = JSON.parse(latestEndpointEvent.attributes.payload);
+  } catch (error) {
+    return policiesTracker;
+  }
+
+  // Get the platform: windows, mac, or linux
+  const hostType =
+    policyHostTypeToPolicyType[
+      endpointPolicyPayload['endpoint-security']?.host?.os?.name as EndpointOSNames
+    ];
+  // Get whether the malware setting for the platform on the most recently provided config is active (prevent or detect is on) or off
+  const userDesiredMalwareState =
+    endpointPolicyPayload['endpoint-security'].Endpoint?.configuration?.inputs[0]?.policy[hostType]
+      ?.malware?.mode;
+
+  // Get the status of the application of the malware protection
+  const malwareStatus =
+    endpointPolicyPayload['endpoint-security'].Endpoint?.policy?.applied?.response?.configurations
+      ?.malware?.status;
+
+  if (!userDesiredMalwareState || !malwareStatus) {
+    // If we get policy information without the mode or status, then nothing to track or update
+    return policiesTracker;
   }
 
-  return updatedEndpointTelemetry;
+  const updatedPoliciesTracker = {
+    malware: { ...policiesTracker.malware },
+  };
+
+  const isAnActiveMalwareState = enabledMalwarePolicyTypes.includes(userDesiredMalwareState);
+
+  // we only check for 'not failure' as the 'warning' state for malware is still technically actively enabled (with warnings)
+  const successfullyEnabled = !!malwareStatus && malwareStatus !== 'failure';
+  const failedToEnable = !!malwareStatus && malwareStatus === 'failure';
+
+  if (isAnActiveMalwareState && successfullyEnabled) {
+    updatedPoliciesTracker.malware.active += 1;
+  } else if (!isAnActiveMalwareState && successfullyEnabled) {
+    updatedPoliciesTracker.malware.inactive += 1;
+  } else if (isAnActiveMalwareState && failedToEnable) {
+    updatedPoliciesTracker.malware.failure += 1;
+  }
+
```

Comment:
Not currently tracking the situation where an endpoint failed to turn the policy off as it's unlikely / not a high priority. Also, since we're in FF, I'm not sure if I can add an additional field to telemetry, @afharo?

Reply:
> since we're in FF, I'm not sure if I can add an additional field to telemetry, @afharo?

If it's a bug, I guess you can add it. If it isn't, you'll have to wait :)